src/serial.rs

```diff
@@ -197,18 +328,43 @@ where
         Tx: TxPin<Usart>,
         Rx: RxPin<Usart>,
     {
+        use self::config::*;
+
+        let config = config.into();
+
+        // Enable USART peripheral for any further interaction.
         Usart::enable_clock(apb);
+        // Disable USART because some configuration bits could only be written
+        // in this state.
+        usart.cr1.modify(|_, w| w.ue().disabled());
 
-        let brr = Usart::clock(&clocks).integer() / baud_rate.integer();
+        let brr = Usart::clock(&clocks).integer() / config.baudrate.integer();
         crate::assert!(brr >= 16, "impossible baud rate");
         usart.brr.write(|w| w.brr().bits(brr as u16));
 
+        let m0 = match config.wordlength {
+            WordLength::DataAndParityBits8 => M_A::BIT8,
+            WordLength::DataAndParityBits9 => M_A::BIT9,
+        };
+
+        let (ps, pce) = match config.parity {
+            Parity::ParityNone => (PS_A::EVEN, PCE_A::DISABLED),
+            Parity::ParityEven => (PS_A::EVEN, PCE_A::ENABLED),
+            Parity::ParityOdd => (PS_A::ODD, PCE_A::ENABLED),
+        };
+
+        usart.cr2.modify(|_, w| w.stop().variant(config.stopbits));
         usart.cr1.modify(|_, w| {
-            w.ue().enabled(); // enable USART
+            w.ps().variant(ps); // set parity mode
+            w.pce().variant(pce); // enable parity checking/generation
+            w.m().variant(m0); // set data bits
```

Comment:
Isn't the configuration for M1 missing, or is it included in `m()`?

Reply:
I's missing in the SVD I was looking into and M0 is named only  M there. :-|